annsa/learning_curve.py

```diff
@@ -13,24 +13,58 @@
-                                           load_pretrained_cae_into_cnn,) 
+                                           load_pretrained_cae_into_cnn,)
+
+
+def training_wrapper(GPU_device_id,
+                     model_id_save_as,
+                     architecture_id,
+                     model_class_id,
+                     testing_dataset_id,
+                     training_dataset_id,
+                     difficulty_setting,
+                     data_augmentation=False,
+                     train_sizes=None,
+                     dense_nodes=None,
+                     weight_id=None,
+                     ):
+    """
+    This function uses source data and background data to generate
+    a random spectrum drawn from a statistical distribution.
+
+    Parameters:
+    -----------
+        GPU_device_id : string
+            GPU device to use for training
+        model_id_save_as : string
+            String used to save resulting model.
+        architecture_id : string
+            The filename of the archetecture to use.
+        model_class_id : string
+            The model class. Examples are 'CNN1D', 'DNN', 'DAE', and 'CAE'
+        testing_dataset_id : string
+            Location of the .npy testing dataset.
+        training_dataset_id : string
+            Location of the .npy training dataset.
+        difficulty_setting : string
+            Predefined dataset difficulty to use. Options are 'easy' or 'full'
+        data_augmentation : Bool
+            True is data augmentation is used, False if not.
+        train_sizes : list, int, optional
+            List of training set sizes to train.
+        dense_nodes : list, int, optional
+            Dense nodes to use for the dense layer
+        weight_id : string
+            Location of pretrained network to use for autoencoder.
+
+    Returns:
+    --------
+        source_spectrum : vector
+            The 1024 length source spectrum
+        background_spectrum : vector
+            The 1024 length background spectrum
+    """
 
-
-def training_wrapper(GPU_device_id = str(0),
-                     model_id_save_as = 'learningcurve-cnn-easy',
-                     architecture_id = '../hyperparameter_search/hyperparameter-search-results/CNN-kfoldseasy_1',
-                     model_class_id = 'CNN1D',
-                     testing_dataset_id = '../../source-interdiction/dataset_generation/validation_dataset_200keV_log10time_1000.npy',
-                     data_augmentation = False,
-
-                        # default 
-                     train_sizes = None,
-                     dense_nodes = None,
-                     weight_id = None,
-                     training_dataset_id = '../../source-interdiction/dataset_generation/testing_dataset_200keV_log10time_10000.npy',
-                     difficulty_setting = 'easy',):
-
-    
     os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
     os.environ["CUDA_VISIBLE_DEVICES"] = GPU_device_id
 
```

Comment:
Shouldn't this code be under:
`if __name__== "__main__":`

Not sure what the best practice is for this, but I like separating the function definitions and executable code.

Reply:
I agree, basically... nothing should be in global scope. If this is a packge, ALL of the global scope stuff could be induced to run as soon as one imports annsa. (yikes.)